### ISSUE TITLE
Error: Timezone database download failed

### ISSUE THREAD

**Issue by decaba** [BestImageViewer/geeqie#1543]
**Setup (please complete the following information):**
 - Distribution: Kubuntu
 - Distribution release: 24.04
 - Geeqie version [geeqie --version]: 2.5+git20241121-c702cce5

**Describe the bug**
When I go to Edit->Preferences->General tab and click the Update button under Timezone database, I get a popup that reads "Error: Timezone database download failed. Operation not supported."

**To reproduce**
See above.

**Expected behavior**

**Screenshots**
![image](https://github.com/user-attachments/assets/e963aa14-f14e-47dc-b80d-66a743b47f81)

**Image sources**

**Error logs or seg. fault files**

**Additional context**
This happened to me while running Geeqie version 2.2 as well.

**Comment by caclark** [comment]
I cannot replicate this problem.

Can you download the file `https://cdn.bertold.org/zonedetect/db/db.zip` via a web browser?

**Comment by decaba** [comment]
Yes, via web browser and via wget. I installed using the geeqie-install-debian.sh script but can try the app image or compile/install from source with --enable-debug-log if that helps. If it's just a matter of putting the db.zip file or its extracted contents someplace that Geeqie can read, I'm fine doing that, too.

**Comment by caclark** [comment]
>but can try the app image or compile/install from source with --enable-debug-log if that helps

Thanks, but there are no debug statements in that part of the code.
Unless I can replicate the problem, at the moment I do not see any way for you to help.
The only difference I see in your setup is that you use Kubuntu - which should be irrelevant.

I am away from home for a few weeks and have only a small laptop with me. When I get back I will try a Kubuntu install.

In the meantime, unzip the file you have downloaded.

Copy `..../db/out_v1/timezone21.bin` to `$HOME/.config/geeqie/timezone21.bin`

When you run Geeqie, the text in the Preferences dialog button should have changed from `Install` to `Update`.

Please leave this open - maybe someone else will have the problem.

**Comment by decaba** [comment]
Thanks. Copying `..../db/out_v1/timezone21.bin` to `$HOME/.config/geeqie/timezone21.bin` does indeed cause the button to change from `Install` to `Update`, but the timestamp does not appear to change anywhere. Also, clicking `Update` causes the same error.

I've tried putting into `$HOME/.config/geeqie/` various combinations of the `timezone21.bin`, `timezone16.bin`, `country16.bin`, and `country21.bin` files (all from the `out_v1` folder), restarting Geeqie each time, but I don't see a difference.

It may be that I am misunderstanding what is supposed to happen. Without the timezone database files in place, the times appear correctly (adjusted for timezone) in the Info sidebar and the Exif/Metadata windows, just not in what I think is called the File Pane. (See screenshot: bits highlighted in yellow are as expected; the bit highlighted in light blue is what I was hoping would change when using the timezone database.) With any combination of the timezone database files in place, nothing changes, other than the button going from `Install` to `Update`.

![Screenshot_20241124_115935](https://github.com/user-attachments/assets/79ed3f36-8b3c-4f12-bbc6-0269b5f5691a)

I wish I could help troubleshoot, but I have no knowledge of C++. Still, if there is something you would like me to try or test on my Kubuntu system, I am happy to do so.

**Comment by caclark** [comment]
The text change from `Install` to `Update` merely indicates that the correct .bin file has been found. Your download problem still exists.

The timezone data appears only in the Info Sidebar and in the Overlay Screen Display. It is described in `https://www.geeqie.org/help/GuideReferenceUTC.html`

The date/time in the files pane is the date/time of the file on the disk.
Using any other date/time leads to the problem that it takes significant time to extract Exif data and convert it - when there are many files in a folder, Geeqie stalls.

I will consider how to improve things.

Exif date/time is already in the cached .sim files. Perhaps local time could be included - but increasing cached data may slow things.

It may be possible to show the Exif/local time in the files display - I had not considered that. 

All the photos I take have Exif data. Therefore the disk file time is meaningless - it is the time I loaded the data from the camera to a computer. There is a plugin to set the disk file time to the file Exif time - I sometimes use that. It could be updated to also use local date/time.
